api/redhatopenshift/HcpCluster/hcpCluster-models.tsp

```diff
@@ -256,13 +267,7 @@ model PlatformProfile {
   outboundType?: OutboundType = OutboundType.loadBalancer;
 
   /** Specifies whether subnets are pre-attached with an NSG */
-  preconfiguredNsgs: boolean;
-
-  /** The id of the disk encryption set to be used for etcd.
-   * Configure this when `etcdEncryption` is set to true
-   * Is used the https://learn.microsoft.com/en-us/azure/storage/common/customer-managed-keys-overview
-  */
-  etcdEncryptionSetId?: string;
+  preconfiguredNsg: boolean;
```

Comment:
What is the purpose of this, and how is it related to the Hypershift CRs? I am trying to understand it better.
I know for example that in the HostedCluster CR there is a "SecurityGroupName" attribute that you pass during creation of an Azure based HostedCluster: https://github.com/openshift/hypershift/blob/main/api/hypershift/v1beta1/hostedcluster_types.go#L1709

What I observed then is that Hypershift ends up setting some rules within it.